lib/policy_machine_storage_adapters/active_record.rb

```diff
@@ -139,45 +139,72 @@ def store_attributes
       end
 
       def descendants(filters = {})
-        assert_valid_filters!(filters)
+        assert_valid_attributes!(filters.keys)
         Assignment.descendants_of(self).where(filters)
       end
 
       def ancestors(filters = {})
-        assert_valid_filters!(filters)
+        assert_valid_attributes!(filters.keys)
         Assignment.ancestors_of(self).where(filters)
       end
 
       def parents(filters = {})
-        assert_valid_filters!(filters)
+        assert_valid_attributes!(filters.keys)
         unfiltered_parents.where(filters)
       end
 
       def children(filters = {})
-        assert_valid_filters!(filters)
+        assert_valid_attributes!(filters.keys)
         unfiltered_children.where(filters)
       end
 
       def link_descendants(filters = {})
-        assert_valid_filters!(filters)
+        assert_valid_attributes!(filters.keys)
         LogicalLink.descendants_of(self).where(filters)
       end
 
       def link_ancestors(filters = {})
-        assert_valid_filters!(filters)
+        assert_valid_attributes!(filters.keys)
         LogicalLink.ancestors_of(self).where(filters)
       end
 
       def link_parents(filters = {})
-        assert_valid_filters!(filters)
+        assert_valid_attributes!(filters.keys)
         unfiltered_link_parents.where(filters)
       end
 
       def link_children(filters = {})
-        assert_valid_filters!(filters)
+        assert_valid_attributes!(filters.keys)
         unfiltered_link_children.where(filters)
       end
 
+      # A series of methods of the form "pluck_from_graph_traversal" which pluck the specified
+      # fields from an element's relatives; returns an array of { attribute => value } hashes.
+      %w(
+        descendants
+        ancestors
+        parents
+        children
+        link_descendants
+        link_ancestors
+        link_parents
+        link_children
+      ).each do |graph_method|
+        define_method("pluck_from_#{graph_method}") do |filters: {}, fields:|
+          assert_valid_attributes!(filters.keys)
+          assert_valid_attributes!(fields)
```

Comment:
This and the rest of these methods may return a pretty cryptic error if `fields` is anything other than a list of objects that respond to `to_sym`. Do we check anywhere that `fields` is appropriately populated?

Reply:
I'm not sure I follow? Most of these PM methods will fail if you pass in random arguments. `assert_valid_attributes!` will just tell you if your fields or filters are actually db columns.